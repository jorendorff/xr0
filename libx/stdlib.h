#ifndef STDLIB_H
#define STDLIB_H

axiom void *
malloc(int size) [ .alloc result; ];

axiom void
free(void *ptr) [
	if (@ptr)
		.dealloc ptr;
	else /* calling free on a non-allocated pointer */
		.undefined;
];

#endif
