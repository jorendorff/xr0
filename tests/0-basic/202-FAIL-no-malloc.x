#include <stdlib.h>

void *p;

void
unit() [ .alloc p; ]
{
	p = NULL;
}
