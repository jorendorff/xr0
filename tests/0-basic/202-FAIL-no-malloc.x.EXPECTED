qed error: did not alloc loc
